### DISCUSSION TITLE
Docker stack doesn't work but standalone container does

### DISCUSSION THREAD

**Discussion by milkyware** [itzg/docker-minecraft-server#1176] [Q&A] [answered]
Hi,

I currently have a docker swarm cluster of 3 Raspberry Pi 4s. I've put together the below compose to try and get the server running, however, when the server is initialising it just seems to stop after **Preparing level "world"** and then fail with **ERROR	mc-server-runner	Failed to stop using rcon-cli**. I also include the logs for both the working standalone container and the failing stack task for comparison.

version: '3.8'

networks:
  default:
  
services:
  server:
    image: itzg/minecraft-server:latest
    environment:
      - TZ-Europe/London
      - EULA=true
      - DEBUG=true
      - DEBUG_EXEC=true
      - MEMORY=2G
    networks:
      - default
    ports:
      - 25565:25565
    deploy:
      mode: replicated
      replicas: 1
      resources:
        limits:
          cpus: "2.0"
          memory: 2G

The main difference I notice is that in the standalone container after **Preparing level "world"** it then continues onto **Preparing start region for dimension minecraft:overworld** which doesn't seem to happen in the stack.

Any help or suggestions would be great. Thank you in advance

[StandaloneLogs.txt](https://github.com/itzg/docker-minecraft-server/files/7689308/StandaloneLogs.txt)
[SwarmLogs.txt](https://github.com/itzg/docker-minecraft-server/files/7689309/SwarmLogs.txt)

**Comment by itzg** [comment, marked as answer]
1.18 is taking even longer to startup than the previous versions, so I'm thinking you will need to configure a longer health check duration. You could set that a lot longer for the initial world creation and then lower it later a bit.

**Reply by milkyware** [reply]
Tried with the healthcheck turned off and it works a treat, thanks. I'll tweak my compose script with a lengthened healthcheck and post it here

**Reply by milkyware** [reply]
My new working compose is below. Thanks again for the help @itzg 

version: '3.8'

networks:
  default:

volumes:
  backups:
  data:
  
services:
  server:
    image: itzg/minecraft-server:latest
    environment:
      - TZ-Europe/London
      - EULA=true
      - ALLOW_NETHER=true
      - ANNOUNCE_PLAYER_ACHIEVEMENTS=true
      - GENERATE_STRUCTURES=true
      - MEMORY=2G
      - SERVER_NAME=Minecraft
      - SPAWN_ANIMALS=true
      - SPAWN_MONSTERS=true
      - SPAWN_NPCS=true
      - TYPE=FORGE
    healthcheck:
      start_period: 10m
    networks:
      - default
    ports:
      - 25565:25565
    volumes:
      - data:/data
    deploy:
      mode: replicated
      replicas: 1
      resources:
        limits:
          cpus: "2.0"
          memory: 2G

**Comment by tehmessiah75** [comment]
Can you share your start up script for the Docker Swarm version please?
Edit: I just want to run a vanilla (not forge) server on a docker swarm cluster

**Reply by milkyware** [reply]
i'm assuming by **startup script** you mean compose file? I think it's the same just minus the `- TYPE=FORGE` environment variable

**Reply by itzg** [reply]
@tehmessiah75 the question has already been answered. 

If you're experiencing what you think is a similar problem then please open a new discussion with logs and your container configuration.
